calyx-ir/src/lib.rs

```diff
@@ -54,3 +54,6 @@ pub mod from_ast;
 
 /// Convinience macros for constructing IR nodes.
 mod macros;
+
+/// Serializer methods for IR nodes.
+pub mod serializers;
```

Comment:
What does an expected use of this module look like for the user. Do they need to import `crate::ir::serializer` in order to use the serialization methods? If so, we should hide the `pub mod ...` using a `cfg` attribute conditioned on the `serialize` feature being enabled. That way, when you forget to enable the feature, the module is not even defined.